internal/app/summon/command/temp_factory.go

```diff
@@ -4,12 +4,10 @@ import (
 	"io/ioutil"
 	"os"
 	"strings"
-
-	"github.com/mitchellh/go-homedir"
 )
 
 // DEVSHM is the location of a memory-mapped directory on Linux.
-const DEVSHM = "/dev/shm"
+var DEVSHM = "/dev/shm"
```

Comment:
Let's give this a meaningful name.  `DEVSHM` is totally confusing when you just see it -- I had no idea what was going on in the tests until I came back and found this comment. 

Side Note: Go conventions dictate [CamelCase for constants](https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps) -- the stdlib follows this as well.

I know this is a `var` and not a `const`, but that appears to have been done only so that it could be changed in the unit tests.  But it actually *is* a constant afaict, and so this seems to be a hack?  Let me know if I've misunderstood this one.

This one already existed I know, but since we're changing this code we should rename `TempFactory` to `TempFileFactory` since that's what it really is. 

Also, importantly, I'd like brief comments explaining the reasons this type (`TempFileFactory`) even exists at the top of the file.  Obviously it creates temp files, but why did we have to create it?  And also why we need to use the memory-mapped directory (is it just for speed?  are there other considerations?).  

More and more, just generally speaking now,  I feel the "why" and the context are extremely important to capture in the code, and having them will significantly improve the experience of future devs.